### ISSUE TITLE
Please provide some documentation around rate limiting

### ISSUE THREAD

**Issue by dancrumb** [rubygems/rubygems.org#1552]
I have a script that pulls data from the RubyGems API.

Every now and again, I get a 429 HTTP Status. This is the "Too Many Request" status.

Unfortunately, the API provides no information about what the rate limit is and how long I should wait before I try again.
In addition, the status message is "Unknown Error", which isn't very developer friendly.

Finally, the documentation doesn't mention rate limiting anywhere.

Some docs and some headers would really help here!

**Comment by segiddins** [comment]
For now, you can take a look at the relevant code -- https://github.com/rubygems/rubygems.org/blob/229caf03312680a17d80dff78665de078b054845/config/initializers/rack_attack.rb, but I don't disagree we should document it

**Comment by shevegen** [comment]
The code could also be simplified perhaps, since it uses 3x the same delay (the 10.minutes part).

The documentation could probably be automatically generated, inserting the current delay in use.

API Documentation is at:

  http://guides.rubygems.org/rubygems-org-api/

Perhaps a very short subsection called "Rate limiting" or something, with one sentence about as to why it happens, and then also mentioning the delay in use.

I agree about the Unknown Error - rubygems sometimes is not too overly specific; I have some problem that is a tiny bit similar in that rubygem refuses to upload some invalid-encoding files (but non .rb files) and it does not exactly tell me which file or files are the guilty party (but I figured it out anyway, it is due to german umlauts and me not using UTF-8 so it's not solely rubygem fault alone here admittedly; I just found the error message really not too helpful in this case, took me several minutes of try-and-error to find out that it was indeed the german umlauts giving problems).

Perhaps the status message error could become officially the "Too Many Request" too, may be easier to find it as well lateron.
